executor/executor.go

```diff
@@ -592,6 +592,23 @@ func init() {
 		if err != nil {
 			return rows, errors.Trace(err)
 		}
+		if ctx.GetSessionVars().EnableChunk {
+			for {
+				chk := exec.newChunk()
+				err = exec.NextChunk(goCtx, chk)
+				if err != nil {
+					return rows, errors.Trace(err)
+				}
+				if chk.NumRows() == 0 {
+					return rows, errors.Trace(exec.Close())
```

Comment:
`exec.Close()` has to be called whatever the execution status is to avoid goroutine leak. Maybe a better way is to add `defer exec.Close()` below `exec.Open()`.